src/RedemptionPool.sol

```diff
@@ -98,74 +128,88 @@ contract RedemptionPool is Ownable {
 
     /// @notice deposit GRO tokens to the pool before the deadline
     /// @param _amount amount of GRO tokens to deposit
-    function deposit(uint256 _amount) external {
-        // Checks that the DEADLINE has not passed
-        if (block.timestamp > DEADLINE) {
-            revert RedemptionErrors.DeadlineExceeded();
-        }
+    function deposit(uint256 _amount) external onlyBeforeDeadline {
+        if (_amount <= 0) revert RedemptionErrors.GreaterThanZeroOnly();
+
         // Transfers the GRO tokens from the sender to this contract
         GRO.safeTransferFrom(msg.sender, address(this), _amount);
         // Increases the balance of the sender by the amount
-        _userBalance[msg.sender] += _amount;
+        _userGROBalance[msg.sender] += _amount;
         // Increases the total deposited by the amount
         totalGRO += _amount;
         // Emits the Deposit event
-        emit Deposit(msg.sender, _amount);
+        emit Deposit(msg.sender, _amount, totalGRO);
     }
 
-    /// @notice withdraw deposited GRO tokens before the deadline
-    /// @param _amount amount of GRO tokens to withdraw
-    function withdraw(uint256 _amount) external {
-        if (block.timestamp > DEADLINE) {
-            revert RedemptionErrors.DeadlineExceeded();
-        }
-        if (_userBalance[msg.sender] > _amount) {
-            revert RedemptionErrors.UserBalanceToSmall();
-        }
-        _userBalance[msg.sender] -= _amount;
-        totalGRO -= _amount;
-        GRO.safeTransfer(msg.sender, _amount);
-        emit Withdraw(msg.sender, _amount);
-    }
-
-    /// @notice Allow to withdraw cUSDC based on amount of GRO tokens deposited per user
-    function claim() external {
-        if (block.timestamp <= DEADLINE) {
-            revert RedemptionErrors.DeadlineExceeded();
-        }
-        if (_userBalance[msg.sender] <= 0) {
+    /**
+     * @notice Allow users to claim their share of USDC tokens based on the amount of GRO tokens they have deposited
+     * @dev Users must have a positive GRO token balance and a non-zero claim available to make a claim
+     * @dev The deadline for making GRO deposits must have passed
+     * @dev Redeems the user's cUSDC tokens for an equivalent amount of USDC tokens and transfers them to the user's address
+     * @dev Decreases the user's claims and contract accounting by the amount claimed
+     */
+    function claim() external onlyAfterDeadline {
+        // Check if the user has a positive GRO token balance
+        if (_userGROBalance[msg.sender] <= 0)
```

Comment:
`(_userGROBalance[msg.sender`  can be stored into local var and used later to avoid multiple SLOADs call to save gas

Reply:
It's not used locally anywhere else in this function? Only loaded again through the getSharesAvailable() call. So would need to add it as a parameter to that function and pass it on from the call here?

Reply:
actually - we can even remove this check completely since function still works with a 0 value